File: backend/app/models/user_pr.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, datetime
import uuid

class UserPrBase(BaseModel):
    user_id: uuid.UUID
    distance: str
    date: date
    time_in_seconds: int
    race_id: Optional[uuid.UUID] = None # Optional link to the race where PR was set

class UserPrCreate(UserPrBase):
    is_official: bool = True # Added field
    race_name: Optional[str] = None # Added field

class UserPrUpdate(BaseModel):
    # Allow updating date, time, or associated race
    # Accept date as string initially for flexibility during parsing
    date: Optional[str] = None
    time_in_seconds: Optional[int] = None
    race_id: Optional[uuid.UUID] = None
    is_official: Optional[bool] = None # Added field
    race_name: Optional[str] = None # Added field

class UserPr(UserPrBase):
    id: uuid.UUID
    created_at: datetime
    is_official: Optional[bool] = True  # Make optional for response handling
    race_name: Optional[str] = None # Added optional field

    class Config:
        from_attributes = True # Use Pydantic v2 standard 

File: frontend/src/types/user_pr.ts
import type { Race } from './race'; // Potentially link back to Race type

// Type definition based on backend/app/models/user_pr.py
export interface UserPr {
  id: string; // UUID comes as string
  user_id: string; // UUID comes as string
  race_id?: string | null; // Optional UUID string or null
  distance: string;
  date: string; // Date string (e.g., "YYYY-MM-DD") from API
  time_in_seconds: number;
  created_at: string; // Timestamp string (ISO 8601) from API
  updated_at?: string; // Add updated_at as optional
  is_official?: boolean | null; // Added: Optional boolean or null for older data
  race_name?: string | null; // Added: Optional string or null

  // Optional: If backend populates related race info
  // race?: Race | null; 
} 